tests/validation/tests/v3_api/test_airgap.py

```diff
@@ -0,0 +1,384 @@
+            bastion_node.host_name,
+            PRIVATE_REGISTRY_USERNAME, PRIVATE_REGISTRY_PASSWORD, cmd)
+    return bastion_node.execute_command(ag_command)
+
+
+def create_nlb_and_add_targets(aws_nodes):
+    # Create internet-facing nlb and grab ARN & dns name
+    lb = AmazonWebServices().create_network_lb(name=HOST_NAME + "-nlb")
+    lb_arn = lb["LoadBalancers"][0]["LoadBalancerArn"]
+    lb_dns = lb["LoadBalancers"][0]["DNSName"]
+    # Create internal nlb and grab ARN & dns name
+    internal_lb = AmazonWebServices().create_network_lb(
+        name=HOST_NAME + "internal-nlb", scheme='internal')
+    internal_lb_arn = internal_lb["LoadBalancers"][0]["LoadBalancerArn"]
+    internal_lb_dns = internal_lb["LoadBalancers"][0]["DNSName"]
+
+    # Upsert the route53 record -- if it exists, update, if not, insert
+    AmazonWebServices().upsert_route_53_record_cname(RANCHER_AG_HOSTNAME,
+                                                     internal_lb_dns)
+
+    # Create the target groups
+    tg80 = AmazonWebServices(). \
+        create_ha_target_group(80, HOST_NAME + "-tg-80")
+    tg443 = AmazonWebServices(). \
+        create_ha_target_group(443, HOST_NAME + "-tg-443")
+    tg80_arn = tg80["TargetGroups"][0]["TargetGroupArn"]
+    tg443_arn = tg443["TargetGroups"][0]["TargetGroupArn"]
+    # Create the internal target groups
+    internal_tg80 = AmazonWebServices(). \
+        create_ha_target_group(80, HOST_NAME + "internal-tg-80")
+    internal_tg443 = AmazonWebServices(). \
+        create_ha_target_group(443, HOST_NAME + "internal-tg-443")
+    internal_tg80_arn = internal_tg80["TargetGroups"][0]["TargetGroupArn"]
+    internal_tg443_arn = internal_tg443["TargetGroups"][0]["TargetGroupArn"]
+
+    # Create listeners for the load balancers, to forward to the target groups
+    AmazonWebServices().create_ha_nlb_listener(
+        loadBalancerARN=lb_arn, port=80, targetGroupARN=tg80_arn)
+    AmazonWebServices().create_ha_nlb_listener(
+        loadBalancerARN=lb_arn, port=443, targetGroupARN=tg443_arn)
+    AmazonWebServices().create_ha_nlb_listener(
+        loadBalancerARN=internal_lb_arn, port=80,
+        targetGroupARN=internal_tg80_arn)
+    AmazonWebServices().create_ha_nlb_listener(
+        loadBalancerARN=internal_lb_arn, port=443,
+        targetGroupARN=internal_tg443_arn)
+
+    targets = []
+
+    for aws_node in aws_nodes:
+        targets.append(aws_node.provider_node_id)
+
+    # Register the nodes to the target groups
+    targets_list = [dict(Id=target_id, Port=80) for target_id in targets]
+    AmazonWebServices().register_targets(targets_list, tg80_arn)
+    AmazonWebServices().register_targets(targets_list, internal_tg80_arn)
+    targets_list = [dict(Id=target_id, Port=443) for target_id in targets]
+    AmazonWebServices().register_targets(targets_list, tg443_arn)
+    AmazonWebServices().register_targets(targets_list, internal_tg443_arn)
+    return lb_dns
```

Comment:
what is the internal lb used for?

Reply:
This is used in conjunction with the Route53 CNAME to set for the Rancher Server URL so that adding clusters will work properly.

The other LB is used to access the URL without setting SOCKS proxy or anything.